sdk/appconfiguration/azappconfiguration/client_api_readonly.go

```diff
@@ -0,0 +1,65 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azappconfiguration
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/appconfiguration/azappconfiguration/internal/generated"
+)
+
+// SetReadOnlyResponse contains the response from SetReadOnly method.
+type SetReadOnlyResponse configurationSettingResponse
+
+// SetReadOnlyOptions contains the optional parameters for the SetReadOnly method.
+type SetReadOnlyOptions struct {
+	// If set to true and the configuration setting exists in the configuration store, update the setting
+	// if the passed-in configuration setting is the same version as the one in the configuration store.
+	// The setting versions are the same if their ETag fields match.
+	OnlyIfUnchanged bool
+}
+
+func (cs ConfigurationSetting) toGeneratedPutLockOptions(ifMatch *azcore.ETag) *generated.AzureAppConfigurationClientPutLockOptions {
+	return &generated.AzureAppConfigurationClientPutLockOptions{
+		IfMatch: (*string)(ifMatch),
+		Label:   cs.label,
+	}
+}
+
+func (cs ConfigurationSetting) toGeneratedDeleteLockOptions(ifMatch *azcore.ETag) *generated.AzureAppConfigurationClientDeleteLockOptions {
+	return &generated.AzureAppConfigurationClientDeleteLockOptions{
+		IfMatch: (*string)(ifMatch),
+		Label:   cs.label,
+	}
+}
+
+// SetReadOnly sets an existing configuration setting to read only or read write state in the configuration store.
+func (c *Client) SetReadOnly(ctx context.Context, setting ConfigurationSetting, isReadOnly bool, options *SetReadOnlyOptions) (SetReadOnlyResponse, error) {
+	var ifMatch *azcore.ETag
+	if options != nil && options.OnlyIfUnchanged {
+		ifMatch = setting.etag
```

Comment:
Should etag be exported? We would want a user to do an unconditional update and remove the etag from a `ConfigurationSetting`

Reply:
That's the logic from other SDKs: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs#L965